### ISSUE TITLE
`prefer-math-trunc`: ignore when LHS is a integer literal

### ISSUE THREAD

**Issue by noftaly** [sindresorhus/eslint-plugin-unicorn#1863]
Hey,

I was creating an enum with bitwise flags like this
```ts
export enum NotificationChannel {
  Mail = 1 << 0,
  SMS = 1 << 1,
  InApp = 1 << 2,
  Push = 1 << 3,
}
```
when suddenly Unicorn gave an error for `1 << 0`.
It is obvious here that i'm not trying to truncate the number, but rather that i'm using a bitwise operator for its main purpose.

I think unicorn should not report this case.
One easy way to solve this is by ignoring when the LHS is an obvious integer literal, or when checking for the specific case of `1<<0` as i don't see another use of `<<0` with another literal number.

I'm not sure if it should be behind an option, because it would not get in the way of people who want the rule, because the rule is "prefer math trunc", implying it is for decimal numbers, not for literal ints.

Maybe a more general option `ignoreForLiterals` should be added, so it would ignore for all literal (ints & floats), as when you inline a literal you generally know what you're doing and what the result will be.

**Comment by fregante** [comment]
I don’t think this is easy to detect. Given that piece of code uses two other generally undesirable features (typescript `enum` and bit-wise operators) I'd just suggest disabling `prefer-math-trunc` as well
